index.js

```diff
@@ -1,27 +1,162 @@
+const connectDB   = require("./config/db") // trae el doc db.js //<------------ LLAMAR A MONGO
+
+require("dotenv").config()            //<------------ LLAMAR A DOTENV
+
+//Body es una libreria de npm que nos ayuda a leer los datos enviados por el cliente a travez de un metodo post o put (req params) - ver recipe controller (npm i body-parser)
+const bodyParser =  require("body-parser") // OJO linea 33 se llama al proyecto
+
+const recipesController     = require("./controllers/recipes.controllers")
+// console.log(recipesController) // <---- {getRecipes: [AsyncFunction: getRecipes], putRecipes: [AsyncFunction: putRecipes]}
+
+//2. MIDDLEWARES----------------------------------------------------------------
+// Es una funcion que se ejecuta despues de recibir una peticion (request) y antes de dar una respuesta
+app.use(express.static("public"))
+
+app.set("views", __dirname + "/views")
+app.set("view engine", "hbs")
+
+hbs.registerPartials(__dirname + "/views/partials")
+
+app.use(bodyParser.urlencoded({extended:true})) //ayuda a leer los datos del cliente
+
+
+connectDB()
+
+//3. RUTAS
+//CRUD
+//Create
+//Read
+//Update
+//Delete
+
+//----------HOME-------------------------------
+app.get("/", (req, res) =>{
+  res.render("index")
+})
+
+//  --------------------------**CREAR**----------------------------
+//  --------------------------**CREAR RECETA**----------------------------
+// una receta cada vez que recargan la pagina
+app.get("/crear-receta", async(req, res) =>{
+
+    const newRecipe = {
+      title: "Mi Fideo Seco",
+      level: "Easy Peasy Lemon Squeezy",
+      ingredients: ["Aceite","fideo","Jitomates","Cebolla","Ajo","Concentrado","Sal","Agua","Queso"],
+      cuisine: "YES",
+      dishType: "main_course",
+      image: "https://cdn7.kiwilimon.com/recetaimagen/28865/29446.jpg",
+      duration: 32,
+      creator: "Rick Astley",
+      created: 2005/05/05
+    }
+
+    const recipeCreated = await Recipe.create(newRecipe)
+
+    console.log(recipeCreated)
+    res.render("create-recipe", {
+      miReci: recipeCreated
+    }) 
+
+})
+
+//  --------------------------**CREAR MULTIPLES RECETA**-------------------(Array)---------
+
+// app.get("/recetas", async (req,res) =>{
+
+//     const allRecipes = await Recipe.insertMany(data)
+//       console.log("Base de Datos poblada con recetas")
+//       res.render("recipes", {
+//         miReci: allRecipes
+//       })
+// })
+
+app.put("/recetas", async (req,res) =>{
+
+  const allRecipes = await Recipe.insertMany(data)
+    console.log("Base de Datos poblada con recetas")
+    res.render("recipes", {
+      miReci: allRecipes
+    })
+})
+
+
+//  --------------------------**Update MULTIPLES RECETA**-------------------(Array)---------
+
+//getRecipes y putRecipes se crearon las funciones en Controllers
+app.get("/todas-recetas", recipesController.getRecipes)
+//id a travez de la url
+// app.put("/:id/update", recipesController.putRecipes)
+
+app.put("/:id/update", async (req,res) =>{
```

Comment:
Aqui no encontre en que ruta muestras el formulario para obtener los datos algo que se me ocurre es que cambiemos el metodo y cada que se acceda a este se cambie la duracion no es lo optimo pero ya vamo a comenzar a ver formularios 😉 

```suggestion
app.get("/:id/update", async (req, res) => {
  const { id } = req.params; // Destruccturacion de objetos (parametros de la url)
  //(datos del formulario body-parser)
  // id que estamos buscando
  await Recipe.findByIdAndUpdate(id, { duration: 100 }, { new: true }); // Documentacion de Mongoos - busca un id y lo modifica //que propiedades vas a
  res.redirect("/todas-recetas");
});```